sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c

```diff
@@ -1026,27 +1067,37 @@ void gnrc_rpl_send_DAO(gnrc_rpl_instance_t *inst, ipv6_addr_t *destination, uint
         return;
     }
     me = &netif->ipv6.addrs[idx];
-
-    /* add external and RPL FT entries */
-    /* TODO: nib: dropped support for external transit options for now */
-    void *ft_state = NULL;
-    gnrc_ipv6_nib_ft_t fte;
-    while (gnrc_ipv6_nib_ft_iter(NULL, dodag->iface, &ft_state, &fte)) {
-        DEBUG("RPL: Send DAO - building transit option\n");
-
-        if ((pkt = _dao_transit_build(pkt, lifetime, false)) == NULL) {
+    if (mop == GNRC_RPL_MOP_NON_STORING_MODE) {
+        /* Add the prefix to include the parent's global IP address. */
+        memcpy(parent, &dodag->dodag_id, 8);
+        if ((pkt = _dao_transit_build(pkt, lifetime, parent, false)) == NULL) {
```

Comment:
[RFC 6550](https://datatracker.ietf.org/doc/html/rfc6550#section-6.7.8):
> A non-storing node that has more than one DAO parent MAY include a Transit 
Information option for each DAO parent as part of the non-storing destination advertisement operation.

Might be out of scope for this PR, but we could consider adding transit options for all parents instead of just the preferred one.